### ISSUE TITLE
Generate different variable names for different generics

### ISSUE THREAD

**Issue by Hu-Wentao** [Milad-Akarie/injectable#38]
If my code looks like this (It's all "Test3", just generic)
```dart
@registerModule
abstract class RegisterModule{
  @preResolve
  Future<Test3<String>> get testThree => Test3.numberThree();
  @preResolve
  Future<Test3<int>> get testThreeInt => Test3.numberThree();
}
```

The following code is generated ( It has two "test3" variables)
I want it to be "test3String" and "test3Int"
```dart
  Future<void> $initGetIt(GetIt g, {String environment}) async {
// ...  some other code ... 
  final test3 = await registerModule.testThree;
  g.registerFactory<Test3<String>>(() => test3);
  final test3 = await registerModule.testThreeInt;
  g.registerFactory<Test3<int>>(() => test3);
  // ...  some other code ... 
}
```

**Comment by Milad-Akarie** [comment]
Hey, @Hu-Wentao hmmm ya that makes sense, but what if we have two types that are registered under name tags. 

```dart
@registerModule
abstract class RegisterModule{
  @Named('test1')
  @preResolve
  Future<Test3<String>> get testThree => Test3.numberThree();
  @Named('test2')
  @preResolve
  Future<Test3<String>> get testThreeInt => Test3.numberThree();
}
```
naming the awaited result test3String will sitll cause a problem.
what if we add a random number at the end would that be weird?

**Comment by Hu-Wentao** [comment]
emmm,   I mean, "injectable" generated two variables with the same name. It's going to look like this: 
(My English is very poor, I hope you can understand it. )
```dart
  @preResolve
  @Named('str')
  Future<Test3<String>> get testThree => Test3.numberThree();

  @preResolve
  @Named('int')
  Future<Test3<int>> get testThreeInt => Test3.numberThree();
```

Two variables named "test3" appear here
```dart
  final test3 = await registerModule.testThree;
  g.registerFactory<Test3<String>>(() => test3, instanceName: 'str');
  final test3 = await registerModule.testThreeInt;
  g.registerFactory<Test3<int>>(() => test3, instanceName: 'int');
  ```

**Comment by Milad-Akarie** [comment]
Yes. I understand the problem. I'm talking about how we can solve it.

**Comment by Hu-Wentao** [comment]
You can call them "Test3String" and "Test3Int", which represent "Test3<String>" and "Test3<int>"
They're going to look like this
```dart
  Future<void> $initGetIt(GetIt g, {String environment}) async {
// ...  some other code ... 
  final test3String = await registerModule.testThree;
  g.registerFactory<Test3<String>>(() => test3);
  final test3Int = await registerModule.testThreeInt;
  g.registerFactory<Test3<int>>(() => test3);
  // ...  some other code ... 
}
```

**Comment by Milad-Akarie** [comment]
@Hu-Wentao fixed in injectable_generator v0.3.3
Thanks for reporting.
